org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/entitytemplates/PropertiesResource.java

```diff
@@ -75,47 +74,62 @@ public Response setProperties(LinkedHashMap<String, String> properties) {
         return RestUtils.persist(this.res);
     }
 
+    // TODO: update description
+
     /**
-     * Gets the defined properties. If no properties are defined, an empty JSON object is returned. If k/v properties
-     * are defined, then a JSON is returned. Otherwise an XML is returned.
+     * Gets the defined properties.
+     *
+     * If no properties are defined, an empty JSON object is returned. If k/v properties are defined, then a JSON is
+     * returned. Otherwise, an empty JSON is returned.
      *
      * @return Key/Value map in the case of Winery WPD mode - else instance of XML Element in case of non-key/value
      * properties
      */
     @GET
     @Produces( {MediaType.APPLICATION_XML, MediaType.TEXT_XML, MediaType.APPLICATION_JSON})
     public @NonNull Response getProperties() {
-        TEntityType tempType = RepositoryFactory.getRepository().getTypeForTemplate(this.template);
-        TEntityTemplate.Properties props = this.template.getProperties();
-        WinerysPropertiesDefinition wpd = tempType.getWinerysPropertiesDefinition();
-        if (props instanceof TEntityTemplate.WineryKVProperties || (props == null && wpd != null)) {
-            if (props == null) {
-                LinkedHashMap<String, String> emptyProps = new LinkedHashMap<>();
-                wpd.getPropertyDefinitions().forEach(propDef -> emptyProps.put(propDef.getKey(), ""));
-
-                TEntityTemplate.WineryKVProperties update = new TEntityTemplate.WineryKVProperties();
-                update.setKVProperties(emptyProps);
-                this.template.setProperties(update);
-                RestUtils.persist(this.res);
-            }
-            Map<String, String> kvProperties = ModelUtilities.getPropertiesKV(template);
-            return Response.ok().entity(kvProperties).type(MediaType.APPLICATION_JSON).build();
-        }
-        // no Winery special treatment, just return the XML properties
-        // These can be null resulting in 200 No Content at the caller
-        if (props == null) {
-            return Response.ok().entity("{}").type(MediaType.APPLICATION_JSON).build();
+        TEntityTemplate.Properties properties = this.template.getProperties();
+        WinerysPropertiesDefinition propertiesDefinition = RepositoryFactory.getRepository()
+            .getTypeForTemplate(this.template)
+            .getWinerysPropertiesDefinition();
+
+        // TODO: apply merge logic to all props types
```

Comment:
We need to discuss this.

Properites are not updated when properties definitions changes. 
In general, this is not a problem, but the frontend renders a form based on the returned properties, and, therefore, does not render new definitions.

Therefore, reconstruct default properties all the time and fill them with already defined values.
Currently, non-defined properties are removed.
This might be something we should not do.